libres/lib/include/ert/concurrency.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef ERT_CONCURRENCY_HPP
+#define ERT_CONCURRENCY_HPP
+
+#include <mutex>
+
+// Standard support for semaphores arrived in C++20, so make our own for now
+// Idea and most of code from
+//  https://raymii.org/s/tutorials/Cpp_std_async_with_a_concurrency_limit.html
+class Semafoor {
```

Comment:
Change name to `Semaphore`?

Reply:
Well, idea was to honour the original source and avoid possible future nameclash. But I'll rename if you insist. :)

Reply:
I can imagine a new member of the team stumbling upon the code and thinking that we are unable to spell the word 'semaphore'. I'm just trying to preempt that discussion, but it's a very very minor issue.

Reply:
Added comment to address this concern  😄